test/dbquery_test/test_pg_generator.py

```diff
@@ -0,0 +1,123 @@
+        self.db.Manipulation("CREATE SCHEMA {}".format(_TEST_SCHEMA))()
+        self.db.Manipulation("SET search_path TO {}".format(_TEST_SCHEMA))()
+
+    def tearDown(self):
+        super(PostgresTestCase, self).tearDown()
+        self.db.Manipulation(
+            "DROP SCHEMA IF EXISTS {} CASCADE".format(_TEST_SCHEMA))()
+        self.db.close()
+
+
+class TestInsertMany(PostgresTestCase):
+    """ Insert several items and test SelectIterator. """
+
+    def test_select_iterator(self):
+        """ Create N rows, use SelectIterator and callback function.
+        """
+        N = 10
+        test_value = "hello"
+        self.db.Manipulation("CREATE TABLE test (id INTEGER, val VARCHAR)")()
+
+        for i in range(N):
+            self.db.Manipulation(
+                "INSERT INTO test VALUES(%s, %s)")(str(i), test_value + str(i))
+
+        # SelectIterator calls this once with a generator of rows.
+        def _callback(row_generator, *args):
+            """ Handles generator which delivers individual rows. Row elements
+                are tuples.
+            """
+            row_counter = 0
+            for row in row_generator:
+                self.assertEqual(
+                    row,
+                    (row_counter, "hello" + str(row_counter)))
+                row_counter += 1
+                self.assertEqual(args[0], "hello")
+                self.assertEqual(args[1], "world")
+            self.assertEqual(row_counter, 10)
+
+        cb_args = ["hello", "world"]
+        arraysize = 2
+        select = self.db.SelectIterator(
+            "SELECT * FROM test", _callback, cb_args, arraysize)
+        sg = select()
+        self.assertEqual(sg, None)
+
+    def test_select_iterator_row_formatter(self):
+        """ Create N rows, use SelectIterator, callback function and formatter.
+        """
+        N = 10
+        test_value = "hello"
+        self.db.Manipulation("CREATE TABLE test (id INTEGER, val VARCHAR)")()
+
+        for i in range(N):
+            self.db.Manipulation(
+                "INSERT INTO test VALUES(%s, %s)")(str(i), test_value + str(i))
+
+        # SelectIterator calls this once with a generator of rows.
+        def _callback(row_generator, *args):
+            """ Handles generator which delivers individual rows formatted with
+                dict_formatter. Row elements are dicts.
+            """
+            row_counter = 0
+            for row in row_generator:
+                self.assertEqual(row["id"], row_counter)
+                self.assertEqual(
+                    row["val"], "hello" + str(row_counter))
+                row_counter += 1
+                self.assertEqual(args[0], "hello")
+                self.assertEqual(args[1], "world")
+            self.assertEqual(row_counter, 10)
+
+        cb_args = ["hello", "world"]
+        arraysize = 2
+        select = self.db.SelectIterator(
+            "SELECT * FROM test", _callback, cb_args, arraysize,
```

Comment:
Same as above, it might be a good idea to specify the ordering so that the test can not fail because of a wrong ordering?